pkg/cmdutil/errors.go

```diff
@@ -68,3 +68,15 @@ func (e NoResultsError) Error() string {
 func NewNoResultsError(message string) NoResultsError {
 	return NoResultsError{message: message}
 }
+
+type UnknownResultsError struct {
+	message string
+}
+
+func (e UnknownResultsError) Error() string {
+	return e.message
+}
+
+func NewUnknownResultsError(message string) UnknownResultsError {
+	return UnknownResultsError{message: message}
+}
```

Comment:
I opted to call this thing `UnknownResultsError` because it seemed the most appropriate with the problem raised in https://github.com/cli/cli/issues/9682#issuecomment-2388360417:

> The branch for the PR might not report any checks if the branch was just created, so before any Checks nor Statuses got registered by external services yet. Re-running the command a few seconds later could show that there are now checks registered as “pending”.

The way I read this is that we don't know, and so it would make sense to exit with something new and unique that is specifically about situations where we don't know.